### PR TITLE
Support both godef and gogetdoc

### PR DESCRIPTION
**Pull request by ramya-rao-a**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -365,6 +375,15 @@
           "type": "boolean",
           "default": true,
           "description": "Autocomplete members from unimported packages."
+        },
+        "go.docsTool": {
+          "type": "string",
+          "default": "godoc",
+          "description": "Pick 'godoc' or 'goreturns' to get documentation. In Go 1.5, godoc is used regardless of the choice here.",
```
Comment (leaxoy) [inline package.json]: Is this a typo mistake? `goreturns` should be `gogetdoc`

Comment (ramya-rao-a) [inline package.json]: Haha, yes. Side effects of Copy+Paste. Can you correct this from your end after merging or you want me to update it?

**Comment by leaxoy** [comment]
You may meet error because `gogetdoc` and `godoc` with `godef` return different function signature.
For eaxample:
gogetdoc: `Println func(args ...interface{}) (n int, err error)`
godoc+godef: `func Println(args ...interface{}) (n int, err error)`

**Comment by leaxoy** [comment]
And if we can get explicit `token` and `package` like `net/http.Response`, then we can use `go doc -u net/http.Response` or `godoc net/http Response` for a more efficiently operation. 
Do you have any idea on this work. @ramya-rao-a

**Comment by ramya-rao-a** [comment]
Oh, I just saw your comments. 

Like I said https://github.com/Microsoft/vscode-go/pull/607#issuecomment-261158324 I made a separate PR #622 that has both our changes.

Can you add your comments there?

**Comment by ramya-rao-a** [comment]
@leaxoy I am not sure what you mean by this comment: https://github.com/leaxoy/vscode-go/pull/1#issuecomment-261135340

**Comment by ramya-rao-a** [comment]
Closing this PR in favor of PR #622

**Comment by leaxoy** [comment]
In `godef` + `godoc` version, we get `package's doc` and then filter `token's doc(like variable, function, struct...)`, if we can get explicit `token` in our source file, then can use `godoc package token` or `go doc package.token` for precise doc, thus no need to find `token's doc` from the whole `package's doc`.

**Comment by ramya-rao-a** [comment]
Agreed, we can continue looking into that

**Pull request closed without merging**